commit 1e449d46f3e44aeb64ab619026e2635d6f282172 (refs/changes/41/1741/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-04-04T11:42:09+02:00 (2 years, 6 months ago)
    
    TOOLS-1739 having a vmapi instance on non-headnode can break 'sdcadm experimental update-other'

diff --git a/lib/cli/do_update_other.js b/lib/cli/do_update_other.js
index 7e33b65..f019750 100644
--- a/lib/cli/do_update_other.js
+++ b/lib/cli/do_update_other.js
@@ -643,13 +643,16 @@ function do_update_other(subcmd, opts, args, cb) {
                     var curImg = parts[parts.length - 2];
                     if (curImg >= '20141030T234934Z') {
                         progress('Running VMAPI migrations');
-                        var argv = [
-                            '/usr/sbin/zlogin',
-                            vmapi.uuid,
-                            'cd /opt/smartdc/vmapi && ./build/node/bin/node ' +
-                                'tools/migrations/add-docker-index.js'
-                        ];
-                        common.spawnRun({argv: argv, log: log}, next);
+                        var cmd = 'cd /opt/smartdc/vmapi && ' +
+                            './build/node/bin/node ' +
+                            'tools/migrations/add-docker-index.js';
+
+                        common.execRemote({
+                            server: vmapi.server_uuid,
+                            vm: vmapi.uuid,
+                            cmd: cmd,
+                            log: log
+                        }, next);
                     } else {
                         return next();
                     }
